### PR TITLE
Making a Tablet and Mobile view

### PR DESCRIPTION
**Pull request by vketrys**
- [x] Turning project to responsive
- [x] Creating custom hook for getting window dimensions
Снимок экрана от 2022-09-28 15-21-22
Снимок экрана от 2022-09-28 15-22-35
Снимок экрана от 2022-09-28 15-23-09
Снимок экрана от 2022-09-28 15-24-59
Снимок экрана от 2022-09-28 15-28-43
Снимок экрана от 2022-09-28 15-29-15

### CODE REVIEW

File: src/components/button/Button.module.scss [hunk 1]
```diff
@@ -39,3 +39,27 @@
...
+    .span {
+      width: 100%;
+    }
+  }
+}
+
+@media only screen and (max-width: $Mobile) {
+  .button {
+    font-size: $Description;
+    width: 170px;
```
Comment (kozlova-anzhelika) [inline src/components/button/Button.module.scss:61]: what about max width for button and flex width? don't fix the width please. if the content changes will you manually change the width? Or will the content pop out of the button?

File: src/components/card/character/CharacterCard.module.scss [hunk 2]
```diff
@@ -84,4 +84,64 @@
       }
     }
   }
+}
+
+@media only screen and (max-width: $SmallTablet) {
+  .CardWrapper {
+    width: 400px;
```
Comment (kozlova-anzhelika) [inline src/components/card/character/CharacterCard.module.scss:91]: use max-width: 400px;

File: src/components/card/episode/EpisodeCard.module.scss [hunk 3]
```diff
@@ -62,4 +62,37 @@
       margin-top: indent(1);
     }
   }
+}
+
+@media only screen and (max-width: $Tablet) {
+  .CardWrapper {
+    width: 250px;
```
Comment (kozlova-anzhelika) [inline src/components/card/episode/EpisodeCard.module.scss:69]: max-width

File: src/components/card/episode/EpisodeCard.module.scss [hunk 4]
```diff
@@ -62,4 +62,37 @@
...
+    height: 275px;
+
+    .CardContent {
+      .Description {
+        a {
+          h2 {
```
Comment (kozlova-anzhelika) [inline src/components/card/episode/EpisodeCard.module.scss:75]: try not to use a tag as a selector

File: src/components/card/location/LocationCard.module.scss [hunk 5]
```diff
@@ -61,4 +61,39 @@
       }
     }
   }
+}
+
+@media only screen and (max-width: $Tablet) {
+  .CardWrapper {
+    width: 250px;
```
Comment (kozlova-anzhelika) [inline src/components/card/location/LocationCard.module.scss:68]: max-width

File: src/components/card/location/LocationCard.module.scss [hunk 6]
```diff
@@ -61,4 +61,39 @@
...
+    height: 250px;
+
+    .CardContent { 
+      .Description {
+        padding: 0;
```
Comment (kozlova-anzhelika) [inline src/components/card/location/LocationCard.module.scss:73]: or padding: unset if you just want to unset previos value

File: src/pages/info/CharacterInfo.tsx [hunk 7]
```diff
@@ -92,7 +94,7 @@ export default function CharacterInfo({
                 next={(): Episode[] => episodes}
                 hasMore
                 loader={<h4>{t('loading')}</h4>}
-                height={450}
+                height={height / 3}
```
Comment (kozlova-anzhelika) [inline src/pages/info/CharacterInfo.tsx:97]: what do you mean "3"?

File: src/pages/info/EpisodeInfo.tsx [hunk 8]
```diff
@@ -60,7 +62,7 @@ export default function EpisodeInfo({ item }: EpisodeInfoProps): JSX.Element {
               next={(): Character[] => characters}
               hasMore
               loader={<h4>{t('loading')}</h4>}
-              height={550}
+              height={height / 2.5}
```
Comment (kozlova-anzhelika) [inline src/pages/info/EpisodeInfo.tsx:65]: what do you mean '2.5'?

File: src/pages/info/LocationInfo.tsx [hunk 9]
```diff
@@ -66,7 +68,7 @@ export default function LocationInfo({ item }: LocationInfoProps): JSX.Element {
               next={(): Character[] => characters}
               hasMore
               loader={<h4>{t('loading')}</h4>}
-              height={550}
+              height={height / 2.5}
```
Comment (kozlova-anzhelika) [inline src/pages/info/LocationInfo.tsx:71]: 2.5?

File: src/components/button/Button.module.scss (hunk 1 above)
Comment (vketrys) [reply src/components/button/Button.module.scss:61]: it's definitely make sense
thank you!

File: src/components/card/episode/EpisodeCard.module.scss (hunk 4 above)
Comment (vketrys) [reply src/components/card/episode/EpisodeCard.module.scss:75]: ok

File: src/pages/info/CharacterInfo.tsx (hunk 7 above)
Comment (vketrys) [reply src/pages/info/CharacterInfo.tsx:97]: this means that scroll component will have height equals 1/3 of full page height

File: src/pages/info/EpisodeInfo.tsx (hunk 8 above)
Comment (vketrys) [reply src/pages/info/EpisodeInfo.tsx:65]: here is 1/4 of page height

File: src/pages/info/LocationInfo.tsx (hunk 9 above)
Comment (vketrys) [reply src/pages/info/LocationInfo.tsx:71]: should I move it to consts or what ?

Comment (kozlova-anzhelika) [reply src/pages/info/LocationInfo.tsx:71]: it would be great) f.e object screenSizeProportions

**Review by kozlova-anzhelika — APPROVED** [review]

File: src/pages/info/EpisodeInfo.tsx (hunk 8 above)
Comment (slamser56) [reply src/pages/info/EpisodeInfo.tsx]: Could we use flex and auto height?

Comment (slamser56) [reply src/pages/info/EpisodeInfo.tsx]: I guess if you want use 1/4 height of your page you can use flex

**Pull request merged**